### PR TITLE
Issue#689 add correlation id in experience events

### PR DESCRIPTION
**Pull request by adobe-activation**
Link to Issue: https://github.com/adobe/xdm/issues/689

### CODE REVIEW

File: schemas/context/experienceevent.schema.json [hunk 1]
```diff
@@ -18,7 +18,13 @@
   "description": "An ExperienceEvent captures observations, including the point in time and identity of the subject involved. Experience Events are fact records of what occurred, representing what happened without aggregation or interpretation. They are critical for time-domain analytics as they allow for observation and analysis of changes that occur in a given window of time and the comparison between multiple windows of time to track trends. Experience Events can be either explicit (directly observable human actions) or implicit (raised without a direct human action).",
   "definitions": {
     "experienceevent": {
-      "properties": { }
+      "properties": {
+        "xdm:correlationID": {
```
Comment (cmathis) [inline schemas/context/experienceevent.schema.json]: This should be named xdm:corrleationId (lower case d) to match the naming format of other ID fields. 

Also, I'm assuming this field is always populated by the data producer before Platform ingestion and not something that happens internally by the Platform so we may want to specify that in the description.

File: schemas/context/experienceevent.schema.json [hunk 2]
```diff
@@ -18,7 +18,13 @@
   "description": "An ExperienceEvent captures observations, including the point in time and identity of the subject involved. Experience Events are fact records of what occurred, representing what happened without aggregation or interpretation. They are critical for time-domain analytics as they allow for observation and analysis of changes that occur in a given window of time and the comparison between multiple windows of time to track trends. Experience Events can be either explicit (directly observable human actions) or implicit (raised without a direct human action).",
   "definitions": {
     "experienceevent": {
-      "properties": { }
+      "properties": {
+        "xdm:correlationId": {
+          "title": "ExperienceEvent Correlation ID",
+          "type": "string",
```
Comment (clubycoder) [inline schemas/context/experienceevent.schema.json]: I would suggest adding `format: "uuid"` to constrain the format, guide users to actually use a UUID, and reduce the chance of users from accidentally sending in junk like "1" or "test".

Comment (prabhum2) [reply schemas/context/experienceevent.schema.json]: Brandon Pack - Do you agree with this recommendation ?

Comment (littlebutty) [reply schemas/context/experienceevent.schema.json]: Yes, I agree with this.  Do we need to be more explicit and say UUID v4?

Comment (adobe-activation) [reply schemas/context/experienceevent.schema.json]: added

File: schemas/context/experienceevent.example.2.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "@id": "https://data.adobe.io/experienceid-2345678",
+  "correlationID": "9fb4e78d-b0bf-4d99-b987-cb37abdbcd15",
```
Comment (Aaronius) [inline schemas/context/experienceevent.example.2.json]: Should this be `correlationId`?

**Comment by littlebutty** [comment]
We asked our external alpha users how they would implement and use this ID and they are very excited about it.  However we've decided to rename it to avoid some confusion.  For now I think wel'' close the PR and submit a new one after we've validated the changes with external users again.

File: schemas/context/experienceevent.schema.json (hunk 1 above)
Comment (raresvla) [reply schemas/context/experienceevent.schema.json]: @cmathis Is this naming convention enforced right now? I ask this because there are quite a few inconsistencies in the base schemas:
* [context/device](https://github.com/adobe/xdm/blob/master/docs/reference/context/device.schema.md): `xdm:typeID` and `xdm:typeIDService`
* [context/search](https://github.com/adobe/xdm/blob/master/docs/reference/context/search.schema.md): `xdm:searchEngineID`
* [context/segmentMembershipItem](https://github.com/adobe/xdm/blob/master/docs/reference/context/segmentmembershipitem.schema.md): `xdm:profileStitchID`, `xdm:segmentID`
* [assets/asset](https://github.com/adobe/xdm/blob/master/docs/reference/assets/asset.schema.md): `xdm:documentID`, `xdm:aliasIDs`
* [data/order](https://github.com/adobe/xdm/blob/master/docs/reference/data/order.schema.md): `xdm:purchaseID`

To sum up, I'd like to understand:
1. Is the `Id` (lower case "d") suffix preferred going forward? Is it also enforced?
2. Is there a plan to make base schemas consistent with this naming convention?

**Comment by cfraser** [comment]
@prabhum2 Please review and merge if it is acceptable.

File: schemas/context/experienceevent.schema.json (hunk 1 above)
Comment (cfraser) [reply schemas/context/experienceevent.schema.json]: The style guide uses `ID` so this field is named correctly.

**Comment by prabhum2** [comment]
Closed this PR as requested by @littlebutty
